### DISCUSSION TITLE
Table field - setting up default column value for ALL rows

### DISCUSSION THREAD

**Discussion by piotrpog** [craftcms/cms#14569] [Ideas]
Right now we can set up default value for columns inside table field. But only for specific number of pre-existing rows.

For example, we want make tabel field have two already existing rows. These table has lightswitch field column. We set up lightswitch value in these rows to "ON" position. But when user adds third row, its lightswitch will be turned off, because its default value of lightswitch.

What i propose is to add ability to set up default values not only to specific number of table rows, but for all rows that user would add.

I specifically meantioned lightswitch earlier, because it would be most usefull with this field existing as column inside table. I recently needed to add ability to temporarily disable rows in table. And each time user added new row, it started as disabled, because by default lightswitch value is set to disabled.

**Comment by chrismlusk** [comment]
I just ran into the same issue: I want to create a table field that has a column with a lightswitch field turned on by default for each new row.

Of course, we can work around it by changing this column heading from `enabled` → `disabled` but that feels like bad AX for users since it's the opposite of how lightswitches are used throughout the platform.

**Comment by piotrpog** [comment]
Maybe this should get moved to discussions?
